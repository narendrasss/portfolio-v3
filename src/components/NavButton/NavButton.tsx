import React from 'react';
import Button, { ButtonProps } from '../base/Button';
import styled from 'styled-components';
import Box from '../base/Box';

const Container = styled(Button)`
  position: fixed;
  top: 0;
  right: 0;
  display: flex;
` as React.FC<ButtonProps>;

const OpenContainer = styled(Container)`
  flex-direction: column;
  align-items: flex-end;
` as React.FC<ButtonProps>;

const CloseContainer = styled(Container)`
  height: 4.5rem;
` as React.FC<ButtonProps>;

const Slice = styled(Box)`
  height: 3px;
  background: black;
`;

const Cross = styled(Slice)`
  position: absolute;
  background: white;
  transform: rotate(45deg);
  &:nth-child(2) {
    transform: rotate(-45deg);
  }
`;

interface NavButtonProps extends ButtonProps {
  isOpen?: boolean;
}

const NavButton: React.FC<NavButtonProps> = ({ isOpen, ...props }) =>
  isOpen ? (
    <CloseContainer
      data-testid="button-close"
      m={1}
      mt="4rem"
      width="4.5rem"
      {...props}
    >
      <Cross width="4.5rem" />
      <Cross width="4.5rem" />
    </CloseContainer>
  ) : (
    <OpenContainer
      data-testid="button-open"
      m={1}
      mt="4rem"
      width="4.5rem"
      {...props}
    >
      <Slice width="4.5rem" mb="1rem" />
      <Slice width="3.6rem" mb="1rem" />
      <Slice width="2.25rem" />
    </OpenContainer>
  );

export default NavButton;
