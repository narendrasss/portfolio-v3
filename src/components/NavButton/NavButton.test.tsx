import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import NavButton from './NavButton';

describe('NavIconButton', () => {
  const mockHandler = jest.fn();

  it('calls click handler when clicked', () => {
    const { getByTestId } = render(<NavButton onClick={mockHandler} />);
    fireEvent.click(getByTestId('button-open'));
    expect(mockHandler).toHaveBeenCalled();
  });

  it('renders open button on initial mount', () => {
    const { queryByTestId } = render(<NavButton onClick={mockHandler} />);
    expect(queryByTestId('button-open')).toBeTruthy();
    expect(queryByTestId('button-close')).toBeNull();
  });

  it('renders close button given isOpen prop', () => {
    const { queryByTestId } = render(
      <NavButton onClick={mockHandler} isOpen />
    );
    expect(queryByTestId('button-close')).toBeTruthy();
    expect(queryByTestId('button-open')).toBeNull();
  });
});
