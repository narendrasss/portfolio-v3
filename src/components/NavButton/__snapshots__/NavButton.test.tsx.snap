// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavIconButton renders without crashing 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa sc-dnqmqq sc-iwsKbI hAPMXe sc-htoDjs jZenbE"
    data-testid="button-open"
    type="button"
    width="4.5rem"
  >
    <div
      class="sc-bdVaJa sc-htoDjs sc-gqjmRU fyxggL"
      width="4.5rem"
    />
    <div
      class="sc-bdVaJa sc-htoDjs sc-gqjmRU ccAPZS"
      width="3.6rem"
    />
    <div
      class="sc-bdVaJa sc-htoDjs sc-gqjmRU kPyfCE"
      width="2.25rem"
    />
  </button>
</DocumentFragment>
`;
